magda-web-client/src/UI/DataPreviewMap.js

```diff
@@ -0,0 +1,153 @@
+    }
+    
+    componentWillUnmount(){
+        window.removeEventListener("message", this.onIframeMessageReceived);
+    }
+
+    componentWillReceiveProps(props){
+        this.createStateFromProps(props);
+    }
+
+    createCatalogItemFromDistribution(){
+        return {
+            "initSources": [
+                {
+                    "catalog": [
+                        {
+                            "name": this.state.selectedDistribution.title,
+                            "type": "magda-item",
+                            "url": "/",
+                            "distributionId": this.state.selectedDistribution.identifier,
+                            "isEnabled": true,
+                            "zoomOnEnable": true
+                        }
+                    ],
+                    "baseMapName": "Positron (Light)"
+                }
+            ]
+        };
+    }
+
+    determineDistribution(props){
+        const distributions = props.dataset.distributions;
+        let dataSourcePreference = props.dataset.dataSourcePreference;
+        if(!dataSourcePreference || !dataSourcePreference.length) dataSourcePreference = defaultDataSourcePreference;
+        dataSourcePreference = dataSourcePreference.map(item => item.toLowerCase());
+        if(!distributions || !distributions.length) return null;
+        let selectedDis = null, perferenceOrder = -1;
+        distributions.forEach(dis => {
+            const format = dis.format.toLowerCase();
+            const disPerORder = dataSourcePreference.indexOf(format);
+            if(disPerORder === -1) return;
+            if(perferenceOrder === -1 || disPerORder<perferenceOrder) {
+                perferenceOrder = disPerORder;
+                selectedDis = dis;
+                return;
+            }
+        });
+        return selectedDis;
+    }
+    
+    onIframeMessageReceived(e){
+        if(this.state.isInitLoading || !this.state.isMapPreviewAvailable || !this.iframeRef) return;
+        const iframeWindow = this.iframeRef.contentWindow;
+        if(iframeWindow !== e.source || e.data !== 'ready') return;
+        iframeWindow.postMessage(this.createCatalogItemFromDistribution(),"*");
+    }
+
+    render(){
+
+        if(this.state.isInitLoading) return (
+            <div className="data-preview-map">
+                <h3>Map Preview</h3>
+                Loading....
+            </div>
+        );
+
+        if(!this.state.isMapPreviewAvailable) return (
+            <div className="data-preview-map">
+                No map preview available
+            </div>
+        );
+
+        return (
+            <div className="data-preview-map">
+                <h3>Map Preview</h3>
+                <div className='mui-row'>
+                <div className='mui-col-sm-12'>
+                    <iframe 
+                        title={this.state.selectedDistribution.title}
+                        width="100%"
+                        height="600px"
```

Comment:
I think height should be 420px

Reply:
@chloeleichen No problem. Will correct that~